### ISSUE TITLE
Rounding Not Defined on TaylorSeries

### ISSUE THREAD

**Issue by jessebett** [JuliaDiff/TaylorSeries.jl#199]
Rounding on `Taylor1{T}` could be defined to return a new `Taylor1` whose coefficients are rounded to the appropriate type.
i.e.
```
t = Taylor1(1.,3) #type Taylor1{Float64}
Float32(t) # Method Error
# expect == Taylor1(Float32(1.),3)
```

**Comment by lbenet** [comment]
Thanks for reporting. 

While the following is not the nicest way, I think it does what you are expecting:
```julia
julia> t = Taylor1(1.,3)
 1.0 + 𝒪(t⁴)

julia> Taylor1(Float32.(t[:]))
 1.0 + 𝒪(t⁴)

julia> typeof(ans)
Taylor1{Float32}
```

Another option is to work from the beginning with coefficients that are `Float32`, simply through:
```julia
julia> t = Taylor1(Float32, 3)
 1.0 t + 𝒪(t⁴)

julia> typeof(ans)
Taylor1{Float32}
```

**Comment by dpsanders** [comment]
It would certainly be nice to be able to do `Float32.(t)` (note the `.`; I don't think it makes sense to define `Float32(t)` ). 

For this I think it's necessary to tap into the broadcasting machinery itself by defining `broadcasted`; see e.g. [here](https://github.com/JuliaIntervals/IntervalArithmetic.jl/blob/9a38d80f5eb67da5d8d721fcc0060ec162c55164/src/multidim/arithmetic.jl#L26) for how it's done for `IntervalBox`.

**Comment by jessebett** [comment]
I'm actually still just chasing down bugs trying to get these types to work through broadcasting of Flux's tracked functions. This was the next bug I ran into, that `Float32` is being called during the evaluation of the `Dense` layer. So while it probably makes sense to have this require `Float32.(t)` it would have the unfortunate effect that I don't think it would play nice with dropping `Taylor1`s where others have other number types.

**Comment by lbenet** [comment]
As @dpsanders mentions, this could be solved by changing the behavior of `eltype` to fix broadcasting; see #198. 

I am currently a bit overloaded, so I hope to find some time to look this up next week; I'm truly sorry. In any case, if you have some progress, let us know.

**Comment by lbenet** [comment]
@jessebett See #200 (branch is `lb/broadcasted`); it fixes the specific case that you described above. Can you play with that to see if it works as expected?

Note that `exp(t)` and `exp.(t)` yield different results; the first one is the actual exponential of a polynomial, while the second one broadcasts `exp` over the elements of `t`.

**Comment by lbenet** [comment]
@jessebett Still working on this with another (more standard?) approach in #207. While rounding still doesn't work yet, some of the issues reported in #200 are working, and some others aren't (including some tests which before used the standard broadcasting machinery).

I guess that `Float32(t)` should simply be equivalent to `convert(Taylor1{Float32}, t)`.

**Comment by lbenet** [comment]
@jessebett Can you take a look into #207 (branch `lb/broadcasting`)? While it still does not solve the original issue here, it does implement broadcasting in a consistent way (tests pass here, as well as those of `TaylorIntegration.jl` and `TaylorModels.jl` using this branch).

Regarding `Float32(t)`, I think the best is that you *locally* overload it, e.g. via `Float32(t::Taylor1) = Taylor1(Float32.(t.coeffs))`. The reason that `Float32(t)` does not yield what you would expect (which is equivalent to the definition above) is because `Taylor1` and `TaylorN` are subtypes of `Number`.

**Comment by lbenet** [comment]
Closing, since this was solved by #207
